src/realm.h

```diff
@@ -1696,6 +1696,15 @@ RLM_API bool realm_list_get_property(const realm_list_t*, realm_property_info_t*
  */
 RLM_API bool realm_list_get(const realm_list_t*, size_t index, realm_value_t* out_value);
 
+/**
+ * Find the value in the list passed as parameter.
+ * @param value to search in the list
+ * @param out_index the index where the value has been found
+ * @param out_found boolean that indicates whether the value is found or not
+ * @return true if no exception occurred.
+ */
+RLM_API bool realm_list_find(const realm_list_t*, const realm_value_t* value, size_t* out_index, bool* out_found);
```

Comment:
Do we need `out_found` - seems like we can easily infer that from `out_index`.

Reply:
Yes we do and I disagree on this proposal. For 2 reasons:
1) finds for set has the same signature (and also most of the find APIs we have in the C API)
2) `out_index` is of type `size_t` this means it can never be negative. If you assign -1 to signal that the value was not found, this index might be `MAX_LIMIT(std::size_t)` which can be a valid index.
I would leave it as it is.

Reply:
Actually, `realm::not_found` is `std::size{-1}`. So in theory, we could get rid of the boolean. However, I would rather keep it as it is to keep it consistent.